### ISSUE TITLE
Incorrect server side handling of ffdhe supported_groups in TLSv1.2 and below

### ISSUE THREAD

**Issue by mattcaswell** [openssl/openssl#21081]
We incorrectly handle ffdhe supported_groups if both the server and the client have configured them - and they are the preferred choice, and TLSv1.2 or below is negotiated.

This can be replicated like this:

````
$ openssl s_server -cert server.pem -groups "ffdhe2048:P-256" -no_tls1_3
$ openssl s_client -groups "ffdhe2048:P-256"
````

This results in the server failing with this error message:

> 40775643657F0000:error:0A0C0103:SSL routines:tls_construct_server_key_exchange:internal error:../ssl/statem/statem_srvr.c:2650:

Seems to affect 3.0 and above.

**Comment by mattcaswell** [comment]
@vdukhovni

**Comment by mattcaswell** [comment]
We don't support ffdhe groups in TLSv1.2. But they are selected anyway - hence the error.

**Comment by vdukhovni** [comment]
@mattcaswell Just in case there's some confusion (deadlock), I did not step forward to author the fix for this. The ball is in your cour (or whoever else steps forward).

**Comment by mattcaswell** [comment]
Yes, I have it scheduled for this sprint - but I'm currently at the committer f2f.

**Comment by mattcaswell** [comment]
Fix in #21274

**Comment by mattcaswell** [comment]
Fixed in master, 3.1 and 3.0. No fix should be needed for 1.1.1. Closing.
